template-parts/content-page.php

```diff
@@ -20,10 +20,10 @@
 		</div>
 
 		<?php
-		$pattern_heading = '/<h2 id="(.*?)">(.*?)<\/h2>/';
+		$pattern_heading = '/<h2 class="(.*?)" id="(.*?)">(.*?)<\/h2>/';
```

Comment:
Are we certain this will *always* have a class?

Reply:
I'm wondering if it would be safer to use the [`WP_HTML_Parser`](https://make.wordpress.org/core/2023/03/07/introducing-the-html-api-in-wordpress-6-2/), to guard against situations where you don't know the full structure.

Alternatively, if you used [named captures](https://php.watch/articles/php-regex-readability#named-captures), this would be a lot more resilient to unexpected markup in the future.

I think between these two, I'd recommend the named captures.
```suggestion
		$pattern_heading = '/<h2[^>]+id="(?<id>.*?)"[^>]*>(?<content>.*?)<\/h2>/';
```
The `$matches` array should now have keys `id` and `content`, which you can reference by name. (Or, since my suggestion does not capture anything before the ID, you can also get the same data via the original indices)

<img width="878" alt="image" src="https://github.com/wikimedia/interconnection-wordpress-theme/assets/442115/7671803c-7d65-434f-99a2-245ad541cfeb">

Reply:
Good question. I know the heading block includes the `wp-block-heading` class, but this was removed at some point and must have made it's way back now.

Reply:
Thank you for the suggestion to use named captures, I updated the code accordingly.